test/tst_shell.cpp

```diff
@@ -79,6 +79,16 @@ private slots:
 		checkStartVars(c);
 	}
 
+	void guiOptionSet() {
+		QStringList args;
+		args << "-u" << "NONE";
+		NeovimConnector *c = NeovimConnector::spawn(args);
+		Shell *s = new Shell(c, ShellOptions());
+		QSignalSpy onOptionSet(s, &Shell::extGuiOptionSet);
+		QVERIFY(onOptionSet.isValid());
+		QVERIFY(SPYWAIT(onOptionSet));
+	}
```

Comment:
Perhaps add another test in guiShimCommands() that checks for the signal when using vim_command

```diff
diff --git a/test/tst_shell.cpp b/test/tst_shell.cpp
index b813494..89d583e 100644
--- a/test/tst_shell.cpp
+++ b/test/tst_shell.cpp
@@ -123,6 +123,12 @@ private slots:
                checkCommand(c, "GuiFont DejaVu Sans Mono:h14:b:l", false);
                QCOMPARE(s->shell()->fontDesc(), QString("DejaVu Sans Mono:h14:l"));
 #endif
+
+               QSignalSpy onOptionSet(s->shell(), &Shell::extGuiOptionSet);
+               checkCommand(c, "call rpcnotify(0, \"Gui\", \"Option\", \"Tabline\", 0)", false);
+               QVERIFY(onOptionSet.isValid());
+               QVERIFY(SPYWAIT(onOptionSet));
+               qDebug() << onOptionSet << onOptionSet.size();
        }
 
 protected:

```